shared/actions/fs/index.js

```diff
@@ -672,6 +672,19 @@ const letResetUserBackIn = ({payload: {id, username}}: FsGen.LetResetUserBackInP
 
 const letResetUserBackInResult = () => undefined // Saga.put(FsGen.createLoadResets())
 
+const deleteFile = (state, action: FsGen.DeleteFilePayload) => new Promise(() => {
+  const opID = Constants.makeUUID()
+  return RPCTypes.SimpleFSSimpleFSRemoveRpcPromise({
+    opID,
+    path: {
+      PathType: RPCTypes.simpleFSPathType.kbfs,
+      kbfs: Constants.fsPathToRpcPathString(action.payload.path),
+    },
+  })
+    .then(() => RPCTypes.SimpleFSSimpleFSWaitRpcPromise({opID}))
```

Comment:
Hmm I'm not sure this would work. The thing that's resolved is passed into `Saga.put`. This would probably need to be a generator function.

Reply:
Seems to work just fine: 
<img width="1048" alt="screen shot 2018-10-17 at 12 51 19 pm" src="https://user-images.githubusercontent.com/59594/47112510-68d40a00-d20b-11e8-84c0-29f804400fa5.png">

Reply:
Oh I think I understand now. This becomes another promise and it just keeps resolving. Sorry for the silly comment!